src/map/lua/luautils.cpp

```diff
@@ -5592,6 +5593,65 @@ namespace luautils
         customMenuContext.erase(PChar->id);
     }
 
+    SendToDBoxReturnCode SendItemToDeliveryBox(std::string const& playerName, uint16 itemId, uint32 quantity, std::string senderText)
+    {
+        uint32 playerID = charutils::getCharIdFromName(playerName);
+        if (!playerID)
+        {
+            return SendToDBoxReturnCode::PLAYER_NOT_FOUND;
+        }
+
+        auto isGil = itemId == 65535;
+
+        // Check to confirm that the item legitimately exists
+        // exclude gil as gil does not have an item pointer
+        auto* PItem = itemutils::GetItemPointer(itemId);
+        if (PItem == nullptr && !isGil)
+        {
+            return SendToDBoxReturnCode::ITEM_NOT_FOUND;
+        }
+
+        // default stack size of gil
+        uint32 stackSize = 999999999;
+        // if not gil then get the actual stack size
+        if (!isGil)
+        {
+            stackSize = PItem->getStackSize();
+        }
+
+        bool quantityMoreThanStackSize = quantity > stackSize;
+
+        // limit the quantity to the stack size of the item
+        quantity = std::clamp<uint32>(quantity, 1, stackSize);
+
+        const char* Query = "INSERT INTO delivery_box (charid, box, itemid, quantity, senderid, sender) VALUES ("
+                            "%u, "     // Player ID
+                            "1, "      // Box ID == 1
+                            "%u, "     // Item ID
+                            "%u, "     // Quantity
+                            "%u, "     // Sender ID ( =Player ID )
+                            "'%s'); "; // Sender Text
+        int32 ret = _sql->Query(Query, playerID, itemId, quantity, playerID, senderText);
+
+        if (ret == SQL_ERROR)
+        {
+            return SendToDBoxReturnCode::QUERY_ERROR;
+        }
+        else
+        {
+            _sql->TransactionCommit();
```

Comment:
Presumably, this should also be paired with `TransactionStart` and potentially `TransactionRollback`, or there's no point in using it?

Reply:
In this case, if the query fired, it's fired. There's no transaction to commit

Reply:
> In this case, if the query fired, it's fired. There's no transaction to commit

Yep, I will add the full transaction logic, thanks! 

Also one question is related to user input escaping, I see existing examples for player name (fixed length of 16) escaping however not for escaping variable length strings (like the sender text). 

I see this closed [PR](https://github.com/LandSandBoat/server/pull/5326/files) that had a fix for the EscapeString function for strings (as it currently does not work). Will the fix be put in another PR? @WinterSolstice8  Thanks!

Reply:
> I see this closed [PR](https://github.com/LandSandBoat/server/pull/5326/files) that had a fix for the EscapeString function for strings (as it currently does not work). Will the fix be put in another PR? @WinterSolstice8 Thanks!

That PR was obsoleted by the new SQL system Zach put in, I believe the escaping is already done for you now.

Reply:
> > I see this closed [PR](https://github.com/LandSandBoat/server/pull/5326/files) that had a fix for the EscapeString function for strings (as it currently does not work). Will the fix be put in another PR? @WinterSolstice8 Thanks!
> 
> That PR was obsoleted by the new SQL system Zach put in, I believe the escaping is already done for you now.

Ah, thanks for the info!

Reply:
> In this case, if the query fired, it's fired. There's no transaction to commit

Ok, I updated the PR to leverage all the appropriate transaction functions now. Also I no longer using `getCharIdFromName` because it requires the name to be in quotes which is inconsistent with other commands that take a player name. Maybe this function and other areas in the code that leverage with similar functionality can be refactored, but out of scope for this PR.